### PR TITLE
fix: KyberDmm

### PR DESCRIPTION
**Pull request by dekz**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinKyberDmm.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        returns (uint256 boughtAmount)
+    {
+        address router;
+        address[] memory pools;
+        address[] memory path;
+        (router, pools, path) = abi.decode(bridgeData, (address, address[], address[]));
+
+        require(pools.length >= 1, "MixinKyberDmm/POOLS_LENGTH_MUST_BE_AT_LEAST_ONE");
+        require(path.length >= 2, "MixinKyberDmm/PATH_LENGTH_MUST_BE_AT_LEAST_TWO");
+        // require(
```
Comment (dekz) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinKyberDmm.sol:72]: I'm dumb and I don't know why this was causing an assertion error?

```
 bridgeData: '0x00000000000000000000000012807818b584a3fa65d38b6c25b13983fe888d6e000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000100000000000000000000000061639d6ec06c13a96b5eb9560b359d7c648c77590000000000000000000000000000000000000000000000000000000000000002000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000defa4e8a7bcba345f687a2f1456f5edd9ce97202'
```

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 2]
```diff
@@ -85,7 +85,7 @@ export const SELL_SOURCE_FILTER_BY_CHAIN_ID = valueByChainId<SourceFilters>(
             ERC20BridgeSource.CryptoCom,
             ERC20BridgeSource.Linkswap,
             ERC20BridgeSource.MakerPsm,
-            ERC20BridgeSource.KyberDmm,
+            // ERC20BridgeSource.KyberDmm,
```
Comment (dekz) [inline packages/asset-swapper/src/utils/market_operation_utils/constants.ts:88]: Disabled until Mainnet deployment

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinKyberDmm.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        internal
+        returns (uint256 boughtAmount)
+    {
+        address router;
+        address[] memory pools;
+        address[] memory path;
+        (router, pools, path) = abi.decode(bridgeData, (address, address[], address[]));
+
+        require(pools.length >= 1, "MixinKyberDmm/POOLS_LENGTH_MUST_BE_AT_LEAST_ONE");
+        require(path.length >= 2, "MixinKyberDmm/PATH_LENGTH_MUST_BE_AT_LEAST_TWO");
```
[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinKyberDmm.sol:71]: or something

**Review by mzhu25 — COMMENTED** [review]
Router address needs to be updated 😅

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 4]
```diff
@@ -915,7 +915,7 @@ export const OASIS_ROUTER_BY_CHAIN_ID = valueByChainId<string>(
 
 export const KYBER_DMM_ROUTER_BY_CHAIN_ID = valueByChainId<string>(
     {
-        [ChainId.Mainnet]: '0x12807818B584a3Fa65D38B6C25B13983fE888D6E',
+        [ChainId.Mainnet]: '0x12807818b584a3fa65d38b6c25b13983fe888d6e',
```
[the remark this answers is not in the document]
Comment (dekz) [reply packages/asset-swapper/src/utils/market_operation_utils/constants.ts:918]: Confirmed with Kyber that they only updated `addLiquidity` function, but we should keep this on the latest we're aware of!

**Pull request merged**